File: medplum-java/medplum-model/src/main/java/com/fhir/ResearchSubject.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableResearchSubject.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ResearchSubject {
  Optional<Meta> meta();

  Optional<String> actualArm();

  String resourceType();

  Optional<Reference> consent();

  Optional<uri> implicitRules();

  Optional<List<Extension>> extension();

  Optional<List<Identifier>> identifier();

  Optional<Period> period();

  Optional<Narrative> text();

  Optional<List<Extension>> modifierExtension();

  Optional<ResearchsubjectStatus> status();

  Reference individual();

  Optional<String> assignedArm();

  Reference study();

  Optional<id> id();

  Optional<code> language();

  Optional<List<ResourceList>> contained();

  static ImmutableResearchSubject.ResourceTypeBuildStage builder() {
    return ImmutableResearchSubject.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/ResearchSubject.ts
import { code } from "./code";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Period } from "./Period";
import { Reference } from "./Reference";
import { ResearchsubjectStatus } from "./ResearchsubjectStatus";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface ResearchSubject {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    status: ResearchsubjectStatus | null | undefined;
    period: Period | null | undefined;
    study: Reference;
    individual: Reference;
    assignedArm: string | null | undefined;
    actualArm: string | null | undefined;
    consent: Reference | null | undefined;
}
